controllers/fenceagentsremediation_controller_test.go

```diff
@@ -194,7 +190,22 @@ var _ = Describe("FAR Controller", func() {
 					conditionStatusPointer(metav1.ConditionTrue))  // SucceededTypeStatus
 				verifyEvent(corev1.EventTypeNormal, utils.EventReasonFenceAgentSucceeded, utils.EventMessageFenceAgentSucceeded)
 				verifyEvent(corev1.EventTypeNormal, utils.EventReasonNodeRemediationCompleted, utils.EventMessageNodeRemediationCompleted)
+			}
+			BeforeEach(func() {
+				node = utils.GetNode("", workerNode)
+				underTestFAR = getFenceAgentsRemediation(workerNode, fenceAgentIPMI, testShareParam, testNodeParam)
 			})
+
+			It("should have finalizer, taint, while the two VAs and one pod will be deleted", testFunc)
+
+			When("remediation is created from escalation remediation supporting same kind template", func() {
+				BeforeEach(func() {
+					underTestFAR.Name = fmt.Sprintf("%s-%s", workerNode, "pseudo-random-test-sufix")
+					underTestFAR.Annotations = map[string]string{"remediation.medik8s.io/node-name": workerNode}
+				})
+				It("should have finalizer, taint, while the two VAs and one pod will be deleted", testFunc)
```

Comment:
It might be better to scaffold the test structure differently. I find it more readable to have a few contexts under `Context("Reconcile"` with `Context("creating valid FAR CR",...)` and maybe Context("creating invalid FAR CR",...)

Moving from 

	Context("Reconcile",..)->
		When("creating valid FAR CR",...)
			It(...)
			When("remediation is created from escalation ...", ...)
				It(...)
.
.
.

To maybe 

	Context("Reconcile",..)->
		Context("creating valid FAR CR",...)
			When("remediation isn't created from escalation ...", ...)
				It(...)
			When("remediation is created from escalation ...", ...)
				It(...)
 .
.
.

Reply:
I've initially constructed it so in order to minimize the diff in order to make the review easier, but I don't mind doing some shuffling.